src/main/scala/rocks/muki/graphql/codegen/ApolloSourceGenerator.scala

```diff
@@ -210,12 +208,12 @@ case class ApolloSourceGenerator(
             val innerSelections = unionSelection.fields.flatMap { field =>
               selectionStats(field, List.empty)
             }
+            val unionTypeName = Type.Name(unionType.name)
+            val unionTermName = Term.Name(unionType.name)
             val params = generateFieldParams(
               unionSelection.fields,
-              typeQualifiers :+ unionName.value
+              typeQualifiers :+ unionTermName.value
```

Comment:
This fixes a bug where the case class field type is incorrect

Reply:
TODO: add a test for this

Reply:
I'm happy if you add a test case, but this is no blocker. So if you can add one nice, if you say you don't have the time, then I'll merge and hope for second PR at some point in time :grin: